bc_obps/compliance/service/elicensing/elicensing_data_refresh_service.py

```diff
@@ -0,0 +1,102 @@
+from dataclasses import dataclass
+from compliance.service.elicensing.elicensing_api_client import ELicensingAPIClient
+from django.db import transaction
+from compliance.models import (
+    ElicensingClientOperator,
+    ElicensingInvoice,
+    ElicensingLineItem,
+    ElicensingPayment,
+    ElicensingAdjustment,
+    ComplianceObligation,
+)
+from datetime import datetime
+from decimal import Decimal
+
+elicensing_api_client = ELicensingAPIClient()
+
+
+@dataclass
+class RefreshWrapperReturn:
+    data_is_fresh: bool
+    invoice: ElicensingInvoice
+
+
+class ElicensingDataRefreshService:
+    """
+    Wrapper for refreshing elicensing data with the compliance_report_version_id
+    """
+
+    @classmethod
+    def refresh_data_wrapper_by_compliance_report_version_id(
+        cls, compliance_report_version_id: int
+    ) -> RefreshWrapperReturn:
+        data_is_fresh = True
+        invoice = ComplianceObligation.objects.get(
+            compliance_report_version_id=compliance_report_version_id
+        ).elicensing_invoice
+        try:
+            ElicensingDataRefreshService.refresh_data_by_invoice(
+                client_operator_id=invoice.elicensing_client_operator_id, invoice_number=invoice.invoice_number  # type: ignore[union-attr]
+            )
```

Comment:
Here, we should either return the updated invoice from this service or call `invoice.refresh_from_db()` after that to ensure we have the latest data. Otherwise, even if `refresh_data_by_invoice` succeeds, we would still be returning the old reference to the invoice object defined on line 34.